lib/plugin.lazy.js

```diff
@@ -0,0 +1,109 @@
+import VueLib from 'vue'
+
+<% if (options.lazy.injectMock) { %>
+const apiMethods = <%= JSON.stringify(options.lazy.mockApiMethods)%>
+let delayedCalls = []
+let SentryMock = {}
+<% } %>
+let resolveSentryReady
+
+export default async function SentryPlugin(ctx, inject) {
+  <% if (options.lazy.injectMock) { %>
+  apiMethods.forEach(key => {
+    SentryMock[key] = (...args) => delayedCalls.push([key, args])
+  })
+
+  window.addEventListener('error', SentryMock.captureException)
+
+  // Inject Sentry mock to the context as $sentry
+  inject('sentry', SentryMock)
+  ctx.$sentry = SentryMock
+  <% } %>
+
+  const loadSentryHook = () => loadSentry(ctx, inject)
+
+  <% if (options.lazy.injectLoadHook) { %>
+  inject('sentryLoad', loadSentryHook)
+  ctx.$sentryLoad = loadSentryHook
+  <% } else { %>
+  window.onNuxtReady(loadSentryHook)
+  <% } %>
+
+  const sentryReady = () => {
+    return new Promise(resolve => {
+      resolveSentryReady = resolve
+    })
+  }
```

Comment:
If `sentryReady()` is called more than once then each call will return a new Promise and only the last one will get resolved. So instead it should be something like:

```js
const sentryReadyPromise = new Promise(resolve => {
  resolveSentryReady = resolve
})

const sentryReady = () => sentryReadyPromise
```

Reply:
Good catch, will fix!